internal/mesh/internal/types/upstreams.go

```diff
@@ -25,7 +25,7 @@ var (
 
 func RegisterUpstreams(r resource.Registry) {
 	r.Register(resource.Registration{
-		Type:     UpstreamsV1Alpha1Type,
+		Type:     UpstreamsType,
```

Comment:
Ah that's fair. In that case, why do we need `UpstreamsType = UpstreamsV1Alpha1Type` assignment above? It was showing as unused for me before. Should it be moved to exports so that someone could refer to the latest version of this type when that's appropriate?

Reply:
Oh I see it's unused because there's no controllers yet. I'll leave it as is.

Reply:
Exactly. Controllers should hopefully be able to only care about the latest version in most cases and allow for resource version translations to handle converting things as necessary. But validations, translation logic, registrations etc. should be specific to versions.